services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/dataaccess/RestrictionInformationManagerImpl.java

```diff
@@ -34,15 +36,17 @@ public RestrictionInformationResponse getRestrictionInformation(UserInfo userInf
 			throw new IllegalArgumentException("Unsupported type: " + request.getRestrictableObjectType());
 		}
 
-		List<UsersRestrictionStatus> statusList = accessRestrictionStatusDao.getSubjectStatus(
-				Arrays.asList(KeyFactory.stringToKey(request.getObjectId())), request.getRestrictableObjectType(),
-				userInfo.getId(), userInfo.getGroups());
+		EntityStateProvider stateProvider = new LazyEntityStateProvider(accessRestrictionStatusDao,
+				usersEntityPermissionsDao, userInfo, KeyFactory.stringToKeySingletonList(request.getObjectId()));
 
-		return statusList.stream().findFirst().map((s) -> {
-			RestrictionInformationResponse info = new RestrictionInformationResponse();
-			info.setHasUnmetAccessRequirement(s.hasUnmet());
-			info.setRestrictionLevel(s.getMostRestrictiveLevel());
-			return info;
+		UserRestrictionStatusWithHasUnmet userRestrictionStatusWithHasUnmet =
+				stateProvider.getUserRestrictionStatusWithHasUnmet((KeyFactory.stringToKey(request.getObjectId())));
+
+		return userRestrictionStatusWithHasUnmet.getUsersRestrictionStatus().getAccessRestrictions().stream().findFirst().map((s) -> {
+					RestrictionInformationResponse info = new RestrictionInformationResponse();
+					info.setHasUnmetAccessRequirement(userRestrictionStatusWithHasUnmet.hasUnmet());
```

Comment:
will use the util.
Missing a test where the most restrictive is on the second AR.